src/pages/Statistics/Statistics.tsx

```diff
@@ -13,21 +31,313 @@ export default function Statistics() {
 
+  const settingsNames = user!.settings.map((e) => e.key);
+  const [radioValue, setRadioValue] = useState(2);
+  const [graphTitle, setGraphTitle] = useState("Weekly Happiness");
+  const [graphSubTitle, setGraphSubTitle] = useState("");
+  const [viewingEntry, setViewingEntry] = useState<Happiness | undefined>(
+    undefined,
+  );
+  const [calCollapsed, setCalCollapsed] = useState<boolean>(false);
+  const [start, setStart] = useState(
+    new Date(
+      new Date().getFullYear(),
+      new Date().getMonth(),
+      new Date().getDate() - 7,
+    ),
+  );
+  const [end, setEnd] = useState(
+    new Date(
+      new Date().getFullYear(),
+      new Date().getMonth(),
+      new Date().getDate(),
+    ),
+  );
+
+  useEffect(() => {
+    // @ts-ignore
+    window.HSOverlay.open(document.querySelector("#show-happiness-modal"));
+  }, [viewingEntry]);
+
+  const { width, height } = useWindowDimensions();
+
+  const {
+    isLoading,
+    data,
+    isError,
+    refetch,
+  }: {
+    isLoading: boolean;
+    data: Happiness[] | undefined;
+    isError: boolean;
+    refetch: (
+      queryFnArgs?: undefined,
+    ) => Promise<Happiness[] | undefined | unknown>;
+  } = useQuery({
+    queryKey: QueryKeys.FETCH_HAPPINESS + " graph query",
+    queryFn: () =>
+      api
+        .get("/happiness/", {
+          start: formatDate(start),
+          end: formatDate(end),
+        })
+        .then((res) => res.data),
+  });
+
+  // Changes selected date range between current week and current month when radioValue variable changes.
+  useEffect(() => {
+    setStart(() => {
+      if (radioValue === 1) {
+        return new Date(
+          new Date().getFullYear(),
+          new Date().getMonth(),
+          new Date().getDate() - new Date().getDay(),
+        );
+      } else {
+        return new Date(new Date().getFullYear(), new Date().getMonth(), 1);
+      }
+    });
+    setEnd(() => {
+      if (radioValue === 1) {
+        return new Date(
+          new Date().getFullYear(),
+          new Date().getMonth(),
+          new Date().getDate() - new Date().getDay() + 6,
+        );
+      } else {
+        return new Date(new Date().getFullYear(), new Date().getMonth() + 1, 0);
+      }
+    });
+    setGraphTitle(() =>
+      radioValue === 1 ? "Weekly Happiness" : "Monthly Happiness",
+    );
+  }, [radioValue]);
+
+  // change subtitle when data changes
+  useEffect(() => {
+    setGraphSubTitle(
+      () =>
+        start.toLocaleString("default", { month: "long" }) +
+        " " +
+        (radioValue === 1
+          ? start.getDate() +
+            "-" +
+            (start.getMonth() === end.getMonth()
+              ? ""
+              : end.toLocaleString("default", { month: "long" }) + " ") +
+            end.getDate()
+          : start.getFullYear()),
+    );
+  }, [data, radioValue, start, end]);
+
+  // react to changes in data
+  useEffect(() => {
+    refetch();
+  }, [data, start, end, graphTitle]);
```

Comment:
shouldn't this only depend on start and end, not data and graphTitle? If it does depend on data and graphTitle, can you explain why?

Reply:
Depending on data is required - if there's cached data or the data updates after the start/end changes, the graph needs to be up to date with the data.

I had it depend on graphTitle as well due to the old version of the graph where the graph would animate to adjust to changes in data, so it was required when switching between the monthly and weekly view (as it otherwise wouldn't change the graph). Since the graph refetches all the time and doesn't animate, graphTitle isn't required. However, I think it can be kept in case we do add the animations back and fix the double loading of the graph when the week/month are changed.

Reply:
Sounds good, thanks!